### PR TITLE
More robust client.

### PR DESCRIPTION
**Pull request by jusa**

### CODE REVIEW

File: src/dbus/clientprivate.cpp [hunk 1]
```diff
@@ -62,14 +64,22 @@ namespace Ngf
...
-{}
+{
+    m_log.setEnabled(QtDebugMsg, false);
+}
 
 Ngf::ClientPrivate::~ClientPrivate()
 {
     disconnect();
+
+    while (!m_events.isEmpty()) {
```
Comment (thp) [inline src/dbus/clientprivate.cpp]: More efficient maybe to just:

```
for (auto &event: m_events) {
    delete event;
}
// m_events gets cleared automatically
```

(or the non-C++11-equivalent of foreach)

File: src/dbus/clientprivate.cpp [hunk 2]
```diff
@@ -134,11 +146,15 @@ void Ngf::ClientPrivate::ngfdStatus(const QString &service, const QString &arg1,
             QDBusInterface *iface = m_iface;
             m_iface = 0;
             // All currently active events are invalid, so clear event list
-            m_events.clear();
-            // Signal that we are disconnected
-            emit q_ptr->connectionStatus(false);
-            if (iface)
+            while (!m_events.isEmpty()) {
```
Comment (thp) [inline src/dbus/clientprivate.cpp]: Same here - just iterate over m_events and `delete` each item, then `.clear()` it.

Comment (thp) [inline src/dbus/clientprivate.cpp]: And if it's a problem that `m_events` might be accessed by multiple threads, you could:

```
QList<Event*> tmp;
std::swap(m_events, tmp);
for (auto &event: tmp) {
    delete event;
}
```

This way, `m_events` will only contain either all alive items or be empty.

File: src/dbus/clientprivate.cpp [hunk 3]
```diff
@@ -249,91 +291,99 @@ void Ngf::ClientPrivate::playPendingReply(QDBusPendingCallWatcher *watcher)
...
+    return changeState(event, StateStopped);
 }
 
-void Ngf::ClientPrivate::removeAt(const int &i)
+void Ngf::ClientPrivate::removeEvent(Event *event)
 {
-    m_events[i].stopped = true;
-    m_events.removeAt(i);
+    if (m_events.removeOne(event))
+        delete event;
```
Comment (thp) [inline src/dbus/clientprivate.cpp]: If `event` is not in `m_events`, maybe log a warning that something is funny?

**Comment by thp** [comment]
Added some comments. If we can use C++11, would be nice to use range-based for loop when going over `m_events` everywhere.

File: src/dbus/clientprivate.cpp (hunk 1 above)
Comment (monich) [inline src/dbus/clientprivate.cpp]: or `qDeleteAll()` which is more portable. There are 100500 ways of doing that.

File: src/dbus/clientprivate.cpp (hunk 3 above)
Comment (jusa) [inline src/dbus/clientprivate.cpp]: Added warning

**Comment by thp** [comment]
LGTM.

**Pull request merged**